app/src/store/data.ts

```diff
@@ -302,8 +302,32 @@ export default function useDataStore() {
     const whitelistUrl = import.meta.env.VITE_GRANT_WHITELIST_URI;
     if (whitelistUrl) {
       const url = whitelistUrl + uniqueStr;
-      const json = await fetch(url).then((res) => res.json());
-      grants = grants.filter((grant) => json[DGRANTS_CHAIN_ID].includes(grant.id));
+
+      const filterWhiteListed = (whiteList: Record<number, [number]>) =>
+        grants.filter((grant) => whiteList[DGRANTS_CHAIN_ID].includes(grant.id));
+
+      const fallback = async () => {
+        const cachedWhiteList: any = await getStorageKey('whitelist');
```

Comment:
We should cast this to `Record<number, number[]>` it seems, instead of typing it as `any`. Might also want to define a new type of `type Whitelist = Record<number, number[]>` in `types/src/app.d.ts`